kasa/exceptions.py

```diff
@@ -1,45 +1,57 @@
 """python-kasa exceptions."""
-from asyncio import TimeoutError
+from asyncio import TimeoutError as _asyncioTimeoutError
 from enum import IntEnum
 from typing import Any, Optional
 
 
-class SmartDeviceException(Exception):
-    """Base exception for device errors."""
+class KasaException(Exception):
+    """Base exception for library errors."""
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        self.error_code: Optional["SmartErrorCode"] = kwargs.get("error_code", None)
-        super().__init__(*args)
 
+class TimeoutError(KasaException, _asyncioTimeoutError):
+    """Timeout exception for device errors."""
 
-class UnsupportedDeviceException(SmartDeviceException):
-    """Exception for trying to connect to unsupported devices."""
+    def __repr__(self):
+        return KasaException.__repr__(self)
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        self.discovery_result = kwargs.get("discovery_result")
-        super().__init__(*args, **kwargs)
+    def __str__(self):
+        return KasaException.__str__(self)
```

Comment:
Would `super()` work here instead of directly naming the `KasaException`?

Reply:
This needs to specify the `KasaException.__str__` otherwise `super()` uses the `asyncio.TimeoutError` implementations which are different from the `KasaException` implementations.  Tests then break as error messages are not as expected.  N.B. this was the case prior to this PR also.